backend/src/packages/chaiNNer_pytorch/pytorch/processing/upscale_image.py

```diff
@@ -55,15 +59,16 @@ def upscale(
             tile_size = NO_TILING
 
         def estimate():
-            element_size = 2 if use_fp16 else 4
-            model_bytes = sum(
-                p.numel() * element_size for p in model.model.parameters()
-            )
+            model_bytes = MODEL_BYTES_CACHE.get(model)
+            if model_bytes is None:
+                model_bytes = sum(p.numel() * 4 for p in model.model.parameters())
+                MODEL_BYTES_CACHE[model] = model_bytes
+            if options.use_fp16:
+                model_bytes = model_bytes // 2
```

Comment:
I think that `if` needs to go inside the cuda block.

Reply:
Oh yeah, true